app/experimenter/experiments/serializers/design.py

```diff
@@ -33,23 +34,47 @@ def update_changelog(self, instance, validated_data):
 
 class PrefValidationMixin(object):
 
-    def validate_pref(self, pref_type, pref_value, field_name):
+    def validate_pref_branch(self, pref_branch):
+        if pref_branch == "Firefox Pref Branch":
+            return {"pref_branch": "Please select a branch"}
+        return {}
+
+    def validate_multi_preference(self, pref):
+        pref_type = pref["pref_type"]
+        pref_value = pref["pref_value"]
+        field_name = "pref_value"
+        errors = {}
+        if pref_type == "Firefox Pref Type":
+            errors["pref_type"] = "Please select a pref type"
+
+        pref_value_error = self.validate_pref_value(pref_type, pref_value, field_name)
+        if pref_value_error:
+            errors[field_name] = pref_value_error
+        return errors
+
+    def validate_pref_value(self, pref_type, pref_value, field_name):
         if pref_type == "integer":
             try:
                 int(pref_value)
             except ValueError:
-                return {field_name: "The pref value must be an integer."}
+                return "The pref value must be an integer."
 
         if pref_type == "boolean":
             if pref_value not in ["true", "false"]:
-                return {field_name: "The pref value must be a boolean."}
+                return "The pref value must be a boolean."
 
         if pref_type == "json string":
             try:
                 json.loads(pref_value)
             except ValueError:
-                return {field_name: "The pref value must be valid JSON."}
-        return {}
+                return "The pref value must be valid JSON."
+
```

Comment:
Have this always return an empty dict or the {field_name: error} dict like the other methods do